src/views/home/components/amount-input/amount-input.view.tsx

```diff
@@ -7,58 +7,65 @@ import Typography from "src/views/shared/typography/typography.view";
 import { Token } from "src/domain";
 
 interface onChangeParams {
-  amount: BigNumber;
+  amount?: BigNumber;
   isInvalid: boolean;
 }
 
 interface AmountInputProps {
+  value?: BigNumber;
   token: Token;
   balance: BigNumber;
   fee: BigNumber;
   onChange: (params: onChangeParams) => void;
 }
 
-const AmountInput: FC<AmountInputProps> = ({ token, balance, fee, onChange }) => {
-  const [value, setValue] = useState("");
-  const classes = useAmountInputStyles(value.length);
-  const actualFee = token.symbol === "WETH" ? fee : BigNumber.from(0);
-
-  const getFixedTokenAmount = (amount: string, decimals: number): string => {
-    const amountWithDecimals = Number(amount) / Math.pow(10, decimals);
-    return Number(amountWithDecimals.toFixed(decimals)).toString();
-  };
+const getFixedTokenAmount = (amount: string, decimals: number): string => {
+  const amountWithDecimals = Number(amount) / Math.pow(10, decimals);
+  return Number(amountWithDecimals.toFixed(decimals)).toString();
+};
 
-  const updateAmountInput = (amount: BigNumber) => {
-    const newAmountWithFee = amount.add(actualFee);
-    const isNewAmountWithFeeMoreThanFunds = newAmountWithFee.gt(BigNumber.from(balance));
-    const isAmountInvalid = isNewAmountWithFeeMoreThanFunds || amount.isZero();
+const AmountInput: FC<AmountInputProps> = ({ value, token, balance, fee, onChange }) => {
+  const defaultInputValue = value ? getFixedTokenAmount(value.toString(), token.decimals) : "";
+  const [inputValue, setInputValue] = useState(defaultInputValue);
+  const classes = useAmountInputStyles(inputValue.length);
+  const actualFee = token.symbol === "WETH" ? fee : BigNumber.from(0);
```

Comment:
I've notice that this check is wrong, it should be: `token.address === ethers.constants.AddressZero`

Reply:
Good catch. I planned to add this comment to your env refactor PR, [where you removed some other hardcoded `weth`](https://github.com/hermeznetwork/bridge-ui/pull/45/files#diff-6f2b6981beaa48a10b9b87095264c4b66a18239e784739e99a2e97cacf1631a5L19).